tests/test_salesforce_incremental_table_reset.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+import datetime
+import pytz
+
+from tap_tester import runner, menagerie, connections
+
+from base import SalesforceBaseTest
+
+
+class SalesforceFullTableReset(SalesforceBaseTest):
+    @staticmethod
+    def name():
+        return "tap_tester_salesforce_full_table_reset"
+
+    @staticmethod
+    def expected_sync_streams():
+        return {
+            'Account',
+            'Contact',
+            'User',
+        }
+
+    @staticmethod
+    def convert_state_to_utc(date_str):
+        """
+        Convert a saved bookmark value of form '2020-08-25T13:17:36-07:00' to
+        a string formatted utc datetime,
+        in order to compare aginast json formatted datetime values
+        """
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return datetime.datetime.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
+    def test_run(self):
+        self.salesforce_api = 'BULK'
+
+        replication_keys = self.expected_replication_keys()
+
+        # SYNC 1
+        conn_id = connections.ensure_connection(self)
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Select only the expected streams tables
+        expected_streams = self.expected_sync_streams()
+        catalog_entries = [ce for ce in found_catalogs if ce['tap_stream_id'] in expected_streams]
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+        streams_replication_methods = {stream: self.FULL_TABLE
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # UPDATE STATE BETWEEN SYNCS
+        new_states = {'bookmarks': dict()}
+        stream_to_current_state = {stream : bookmark.get(self.expected_replication_keys()[stream].pop())
+                                   for stream, bookmark in first_sync_bookmarks['bookmarks'].items()}
+        for stream in stream_to_current_state:
+            replication_key = list(replication_keys[stream])[0]
+            # Remove stream User to reset
+            if stream != 'User':
+                new_states['bookmarks'][stream] = {replication_key: stream_to_current_state[stream]}
+        menagerie.set_state(conn_id, new_states)
+
+        # SYNC 2
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        #Verify if the 2 syncs returned the same set of records
```

Comment:
Comment does not match the assertion.  One mentions a record comparison while the other does a bookmark comparison.  Also the bookmark comparison is unfiltered so it can fail if new data comes in between syncs.